src/iceberg_destination.rs

```diff
@@ -0,0 +1,201 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use arrow::array::RecordBatch;
+use arrow_schema::{Field, Schema, SchemaRef};
+use futures::{pin_mut, StreamExt, TryStream};
+use iceberg::io::FileIO;
+use iceberg::spec::{
+    DataContentType, DataFileFormat, FormatVersion, Manifest, ManifestContentType, ManifestEntry,
+    ManifestFile, ManifestListWriter, ManifestMetadata, ManifestStatus, ManifestWriter, Operation,
+    PartitionSpec, Snapshot, Struct, Summary, TableMetadata, TableMetadataBuilder,
+};
+use iceberg::writer::file_writer::location_generator::{
+    DefaultFileNameGenerator, DefaultLocationGenerator,
+};
+use iceberg::writer::file_writer::{FileWriter, FileWriterBuilder, ParquetWriterBuilder};
+use iceberg::TableCreation;
+use log::info;
+use parquet::arrow::PARQUET_FIELD_ID_META_KEY;
+use parquet::file::properties::WriterProperties;
+use url::Url;
+use uuid::Uuid;
+
+use crate::error::DataLoadingError;
+
+pub async fn record_batches_to_iceberg(
+    record_batch_stream: impl TryStream<Item = Result<RecordBatch, DataLoadingError>>,
+    arrow_schema: SchemaRef,
+    target_url: Url,
+) -> Result<(), DataLoadingError> {
+    pin_mut!(record_batch_stream);
+
+    let mut file_io_props: Vec<(String, String)> = vec![];
+    if let Ok(aws_endpoint) = std::env::var("AWS_ENDPOINT") {
```

Comment:
```suggestion
    if let Ok(aws_endpoint) = std::env::var(&AmazonS3ConfigKey::Endpoint) {
```

Reply:
That wouldn't work, `AmazonS3ConfigKey::Endpoint.as_ref()` is "aws_endpoint" rather than "AWS_ENDPOINT"

Reply:
Do a `.to_uppercase()` in this case.

Reply:
That would work, but it makes me think the coincidence is accidental and AmazonS3ConfigKey is not designed to correspond to environment variables in this way.

Reply:
It got me thinking that that `AmazonS3ConfigKey::Endpoint` might map just to `endpoint`, and not `aws_endpoint`. So, better to leave a string as you did.